runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/request/HttpRequestBuilder.kt

```diff
@@ -113,7 +102,7 @@ public suspend fun dumpRequest(request: HttpRequestBuilder, dumpBody: Boolean):
     val buffer = SdkBuffer()
 
     // TODO - we have no way to know the http version at this level to set HTTP/x.x
-    buffer.writeUtf8("${request.method} ${request.url.encodedPath}\r\n")
+    buffer.writeUtf8("${request.method} ${request.url.requestRelativePath}\r\n")
```

Comment:
**question**: Why would this (and elsewhere where we've replaced `encodedPath` with `requestRelativePath`) not be `path.encoded`?

Reply:
The previous `encodedPath` field was a misnomer. Beyond just the path, it also included the query parameters and fragment. Further, it also prepended a `/` if none was present. Replacing it with `path.encoded` (as I did initially based on the name) causes breaks in signing, protocol tests, and several other places.

I chose to name it `requestRelativePath` because the places this is used generally deal with forming an HTTP request line (e.g., `GET /requestRelativePath?foo=bar#baz HTTP/1.1`). It's still kinda wordy though...open to alternative naming suggestions.